components/CategoriesBar/lib/useRect.ts

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import { useResizeObserver } from '@react-hookz/web';
+import { type RefObject, useState } from 'react';
+
+interface RectResult {
+    bottom: number;
+    height: number;
+    left: number;
+    right: number;
+    top: number;
+    width: number;
+}
+
+export function useRect<T extends HTMLElement>(ref: RefObject<T>) {
+    const [rect, setRect] = useState<RectResult>(
+        ref && ref.current ? getRect(ref.current) : getRect(),
+    );
+
+    useResizeObserver(ref, () => {
+        if (ref.current) {
+            setRect(getRect(ref.current));
+        }
+    });
```

Comment:
This also fixes the resizing issue that @oleg-semyonov mentioned in the other PR: https://github.com/prezly/theme-nextjs-lena/pull/489#pullrequestreview-1389936700


https://github.com/prezly/theme-nextjs-lena/assets/67554982/b468090d-c54c-4a65-9533-93bdd1684137